pkg/operations/switchover_util.go

```diff
@@ -117,31 +117,41 @@ func checkPodRoleLabelConsistency(ctx context.Context,
 	return false, nil
 }
 
-// getServiceableNWritablePod returns the serviceable and writable pod of the component.
-func getServiceableNWritablePod(ctx context.Context, cli client.Reader, synthesizeComp component.SynthesizedComponent) (*corev1.Pod, error) {
-	if synthesizeComp.Roles == nil {
-		return nil, errors.New("component does not support switchover")
-	}
-
-	targetRole := ""
-	for _, role := range synthesizeComp.Roles {
-		if role.Serviceable && role.Writable {
-			if targetRole != "" {
-				return nil, errors.New("component has more than role is serviceable and writable, does not support switchover")
-			}
-			targetRole = role.Name
-		}
-	}
-	if targetRole == "" {
-		return nil, errors.New("component has no role is serviceable and writable, does not support switchover")
+// get the Pod object whose current role needs to be transferred to another pod
+func getPodToPerformSwitchover(ctx context.Context, cli client.Reader, synthesizedComp *component.SynthesizedComponent) (*corev1.Pod, error) {
+	role, err := getTargetRoleName(synthesizedComp.Roles)
+	if err != nil {
+		return nil, err
 	}
+	pod, err := getPodByRole(ctx, cli, synthesizedComp, role)
+	return pod, err
+}
 
+func getPodByRole(ctx context.Context, cli client.Reader, synthesizeComp *component.SynthesizedComponent, targetRole string) (*corev1.Pod, error) {
 	pods, err := component.ListOwnedPodsWithRole(ctx, cli, synthesizeComp.Namespace, synthesizeComp.ClusterName, synthesizeComp.Name, targetRole)
 	if err != nil {
 		return nil, err
 	}
 	if len(pods) != 1 {
-		return nil, errors.New("component pod list is empty or has more than one serviceable and writable pod")
+		return nil, errors.New("target pod list is empty or has more than one pod")
 	}
 	return pods[0], nil
 }
+
+// getTargetRole returns the role on which the switchover is performed
+// FIXME: the assumption that only one role supports switchover may change in the future
+func getTargetRoleName(roles []appsv1.ReplicaRole) (string, error) {
```

Comment:
We don't need to infer the target pods, just let the user specify the specific roles or pods.